pkg/crypto/tinkcrypto/primitive/composite/ecdhes/ecdhes_key_template.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ecdhes
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/google/tink/go/aead"
+	commonpb "github.com/google/tink/go/proto/common_go_proto"
+	tinkpb "github.com/google/tink/go/proto/tink_go_proto"
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle"
+	ecdhespb "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/proto/ecdhes_aead_go_proto"
+)
+
+// ECDHES256AES256GCMKeyTemplate is a KeyTemplate that generates an ECDH-ES P-256 key wrapping and AES256-GCM CEK with
```

Comment:
Why is 256 repeated twice?

Reply:
Because the key wrapping uses P-256 keys and content encryption uses AES256GCM primitive. 

We can add P-128 key wrapping and AES256GCM too where the template would be `ECDHES128AES256GCMKeyTemplate`

Reply:
Are you missing the "P"? "256" is ambiguous when you mean `P-256` / `secp256r1`.

What's `P-128` key wrapping mean?

Reply:
I think ECDH is for elliptic curves only, the `P` is assumed here. 

> What's `P-128` key wrapping mean? 
I meant AES128GCM primitive.
Go Jose calls the ECDH-ES key wrapping algorithms this way:
```
ECDH_ES_A128KW     = KeyAlgorithm("ECDH-ES+A128KW")     // ECDH-ES + AES key wrap (128)
	ECDH_ES_A192KW     = KeyAlgorithm("ECDH-ES+A192KW")     // ECDH-ES + AES key wrap (192)
	ECDH_ES_A256KW     = KeyAlgorithm("ECDH-ES+A256KW")     // ECDH-ES + AES key wrap (256)
```

I think the AES key wrap is unnecessary, ECDH-ES is sufficient (ie key derivation only). In other words, this call in [wrap()](https://github.com/hyperledger/aries-framework-go/blob/1c70c234ef1f0b0edfe779394292c9208ccaddee/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle/ecdhes_concatkdf_sender_kw.go#L48) function:
`wk, err := josecipher.KeyWrap(block, s.cek)`
and this call in [unwrap()](https://github.com/hyperledger/aries-framework-go/blob/1c70c234ef1f0b0edfe779394292c9208ccaddee/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle/ecdhes_concatkdf_recipient_kuw.go#L59) function:
`return josecipher.KeyUnwrap(block, recWK.EncryptedCEK)` 
should be replaced by `block.Encrypt(s.cek, nil)` and `block.Decrypt(recK.EncryptedCEK, nil)` respectively.

This means I need to change the default key wrapping algorithm [ECDH-ES+A256KW](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle/ecdhes_concatkdf_sender_kw.go#L19) to just `ECDH-ES`, but as far as Go Jose is concerned, there is no key wrapping with `ECDH-ES` as mentioned [here](https://github.com/square/go-jose/blob/master/asymmetric.go#L339). To follow Go Jose, I've used `ECDH-ES+A256KW` as the only key wrapping algorithm in our new Tink primitives. 

But we should probably do only key derivation and cek encryption/decryption with the derived kek without calling josesipher.KeyWrap()/KeyUnrwap(). What do you think @troyronda ?

Reply:
P is not assumed. there is also x25519, for example.

Reply:
added P in the template function name

Reply:
There's been some lint errors.. so for now I can't add a P... but 25519 key templates will be different.